airs2icarol/airs2icarol.py

```diff
@@ -315,133 +329,165 @@ def _translated_type_xpath(contact_type, xpath, root_element, joinstr=u'; ', _=N
+        u'PhoneNumberAfterHours': partial(_xpath_join, 'Phone[./Type/text() = "Voice" and ./Description/text() = "After Hours"]/PhoneNumber/text()'),
+        u'PhoneNumberBusinessLine': partial(_xpath_join, 'Phone[./Type/text() = "Voice" and ./Description/text() = "Office"]/PhoneNumber/text()'),
+        u'PhoneFax': partial(_xpath_join, 'Phone[./Type/text() = "Fax"]/PhoneNumber/text()'),
+        u'PhoneTTY': partial(_xpath_join, 'Phone[./Type/text() = "TTY/TDD"]/PhoneNumber/text()'),
+
+        u'WebsiteAddress': partial(_xpath_join, 'URL/Address/text()'),
+        u'EmailAddressMain': partial(_xpath_join, 'Email/Address/text()'),
+
+        u'CoverageArea': partial(_xpath_join, 'GeographicAreaServed/Description/text()'),
+
+        u'MainContactTitle': partial(_translated_type_xpath, _('Primary Contact'), 'Contact[@Type = "%s"]/Title/text()'),
+        u'MainContactName': partial(_translated_type_xpath, _('Primary Contact'), 'Contact[@Type = "%s"]/Name/text()'),
+        u'MainContactPhoneNumber': partial(_translated_type_xpath, _('Primary Contact'), 'Contact[@Type = "%s"]/Phone/PhoneNumber/text()'),
+        u'MainContactEmailAddress': partial(_translated_type_xpath, _('Primary Contact'), 'Contact[@Type = "%s"]/Email/Address/text()'),
+
+        u'LastVerifiedOn': partial(_xpath_join, 'ResourceInfo/@DateLastVerified'),
+        u'LastVerifiedByName': partial(_xpath_join, 'ResourceInfo/Contact/Name/text()'),
+        u'LastVerifiedByTitle': partial(_xpath_join, 'ResourceInfo/Contact/Title/text()'),
+        u'LastVerifiedByEmailAddress': partial(_xpath_join, 'ResourceInfo/Contact/Email/Address/text()'),
+        u'LastVerifiedByPhoneNumber': partial(_xpath_join, 'ResourceInfo/Contact/Phone[Type/text() = "Voice"]/PhoneNumber/text()'),
+
+        u'FeeStructureSource': partial(_xpath_join, 'FeeStructure/text()'),
+        u'ApplicationProcess': partial(_xpath_join, 'ApplicationProcess/Description/text()'),
+        u'LanguagesOffered': _languages_offered,
+        u'InternalNotesForEditorsAndViewers': partial(_xpath_join, 'InternalNote/text()'),
+	u'InternalNote': partial(_xpath_join, 'EditorsNote/text()'),
+	u'Custom_PublicComments': partial(_xpath_join, 'PublicNote/text()'),
+        u'Eligibility': _eligibility,
+        u'HoursOfOperation': partial(_xpath_join, 'TimeOpen/Notes/text()'),
+
+    }
 
 
 }
+###This was moved from the convert_xml function to be global in order to access it in _convert_part. There may be better solutions
+header = ['AgencyNUM', 'NUM', 'Record Type', 'CultureCode']
+
+#Get all keys from dictionary
+###Done quickly, can probably remove one loop
+field_header = []
+for entry in _airs_type_mapping:
+    for key in _airs_type_mapping[entry]:
+        field_header.append(key)
+
+#Remove duplicates
+###Sorted is unnecessary for functionality, however having the fields in alphabetical order is more appealing.
+field_header = sorted(list(set(field_header)))
```

Comment:
Okay I understand about the fields that need to be at the end. Use a key function parameter to ```sorted``` to change the sort order. The fn will take individual values from the field_header to be sorted and return a tuple of (val in custom_fields, val).